#! /bin/sh /usr/share/dpatch/dpatch-run
## 26_dcm_memleak_fix.dpatch by  <balint@balintreczey.hu>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Fix memory leak in DCM dissector

@DPATCH@

Index: trunk-1.0/epan/dissectors/packet-dcm.c
===================================================================
--- trunk-1.0/epan/dissectors/packet-dcm.c	(revision 28410)
+++ trunk-1.0/epan/dissectors/packet-dcm.c	(revision 28411)
@@ -289,15 +289,10 @@
 {
     dcmState_t *ds;
 
-    if (NULL == (ds = (dcmState_t *) g_malloc(sizeof(dcmState_t)))) {
+    if (NULL == (ds = (dcmState_t *) se_alloc0(sizeof(dcmState_t)))) {
 	return NULL;
     }
-    ds->pdu = 0;
-    ds->tlen = ds->rlen = 0;
     ds->valid = TRUE;
-    memset(ds->orig, 0, sizeof(ds->orig));
-    memset(ds->targ, 0, sizeof(ds->targ));
-    memset(ds->resp, 0, sizeof(ds->resp));
     ds->first = ds->last = NULL;
     return ds;
 }
